spec/notification_center_spec.rb

```diff
@@ -0,0 +1,484 @@
+        expect(notification_center.add_notification_listener(
+         'Test notification type',
+         @callback_reference
+        )).to eq(nil)
+        expect(spy_logger).to have_received(:log).once
+                               .with(Logger::ERROR, 'Invalid notification type.')
+      end
+      
+      it 'should log and return nil if invalid callable given' do
+        expect(notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE],
+         'Invalid callback!'
+        )).to eq(nil)
+        expect(spy_logger).to have_received(:log).once
+           .with(Logger::ERROR, 'Invalid notification callback given.')
+      end
+    end
+    
+    describe 'test add notification with valid type and callback' do
+      it 'should add, and return notification ID when a plain function is passed as an argument ' do
+        expect(notification_center.add_notification_listener(
+         Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE],
+         @callback_reference
+        )).to eq(1)
+        # verifies that one notification is added
+        expect(notification_center.notifications[Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE]].length)
+         .to eq(1)
+      end
+    end
+    
+    describe 'test add notification for multiple notification types' do
+      it 'should add, return notification ID and validate callbacks when a valid callback is added for each notification type ' do
+        Optimizely::NotificationCenter::NOTIFICATION_TYPES.values.each_with_index {|value, index|
+          notification_id = index+1
+          expect(notification_center.add_notification_listener(
+           value,
+           @callback_reference
+          )).to eq(notification_id)
+        }
+        notification_center.notifications.each do |key, _array|
+          notification_center.notifications[key].each do |notification|
+            expect(notification[:callback]).to eq @callback_reference
+          end
+        end
+      end
+      
+      it 'should add, return notification ID and validate callbacks when multiple valid callbacks are added for a single notification type' do
+        class CallBackSecond
+          def call(_args)
+            'Test multi listner.'
+          end
+        end
+  
+        @callback_second = CallBackSecond.new
+        @callback_reference_second = @callback_second.method(:call)
+        
+        notification_type = Optimizely::NotificationCenter::NOTIFICATION_TYPES[:ACTIVATE]
+
+        expect(notification_center.add_notification_listener(
+          notification_type,
+         @callback_reference)
+        ).to eq(1)
+        
+        expect(notification_center.add_notification_listener(
+          notification_type,
+         @callback_reference_second)
+        ).to eq(2)
+        
+        expect(notification_center.notifications[notification_type][0][:callback]).to eq @callback_reference
```

Comment:
Do not compare callbacks just assert that notifications length for ACTIVATE is now 2